hicommonwealth/commonwealth
# Update pagination (API + UI)

## Link to Issue
Closes: #4455
Closes: #4525 
Closes: #4573

## Description of Changes
- Affected areas:
  - Search bar
  - Search page (all tabs)
  - Admin manage community (upgrade member)
  - Admin members search
  - Quill mentions member search
  - Thread collaborators modal
- Removes the minimum length for search terms (better UX), and sets max results per page to 100
- Add more explicit typing in a few areas where it was lacking
- Removes unused route `createReaction`

## Test Plan
- CA testing the affected areas:
  - From home page, enter a search query into the search bar– confirm that results show as expected
  - Submit the query– on the search results page, confirm that all tabs show results as expected + Sort By works correctly
  - As admin, navigate to a community– go to the Manage community page and confirm that the member search box works correctly + upgrading members works as expected
  - As admin, navigate to the Members page, confirm that search + pagination works correctly
  - Go to any page with a quill editor– enter the `@` symbol and start typing a username– confirm that a list of suggested members shows up correctly
  - Go to any thread, confirm that the collaborators modal allows searching for users as expected

## Deployment Plan
N/A

## Other Considerations
- Mainly need help testing the admin upgrade form
